site/it/community/contribute/code_style.md

```diff
@@ -0,0 +1,166 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+```
+
+Usare `six` per scrivere codice compatibile (per esempio, `six.moves.range`).
+
+
+## Stile di codifica C++
+
+Le modifiche al codice C++ di TensorFlow dovranno essere conformi alla [Google C++ Style
+Guide](https://google.github.io/styleguide/cppguide.html). Usare `clang-tidy` per 
+controllare le vostre modifiche C/C++.
+
+Per installare  `clang-tidy` su Ubuntu 16+:
+
+
+```bash
+$ apt-get install -y clang-tidy
+```
+
+Potete controllare un file C/C++ usando:
+
+```bash
+$ clang-format <my_cc_file> --style=google > /tmp/my_cc_file.cc
+$ diff <my_cc_file> /tmp/my_cc_file.cc
+```
+
+## Altri linguaggi
+
+*   [Google Java Style Guide](https://google.github.io/styleguide/javaguide.html)
+*   [Google JavaScript Style Guide](https://google.github.io/styleguide/jsguide.html)
+*   [Google Shell Style Guide](https://google.github.io/styleguide/shell.xml)
+*   [Google Objective-C Style Guide](https://google.github.io/styleguide/objcguide.html)
+
+
+
+
+## Convenzioni e usi particolari in TensorFlow
+
+### Tensori
+
+*   Le operazioni che trattano batch possono assumere che la **prima dimensione** di
+    un Tensore sia la dimensione del batch.
+*   Nella maggioranza del modelli, l' **ultima dimensione** è il numero dei  _canali_.
+*   Dimensioni ad eccezione della prima e dell'ultima, di solito, rappresentano le dimensioni dello _spazio_:
+    la lunghezza della sequenza, o la dimensione dell'immagine.
+*   Quando è possibile, usare un operatore sovraccaricato del Tensore, invece di una funzione 
+    TensorFlow. Per esempio, noi preferiamo `**`, `+`, `/`, `*`, `-`, `and/or` a
+    `tf.pow`, `tf.add`, `tf.divide`, `tf.multiply`, `tf.subtract`, e `tf.logical_*` —
+    a meno che si desideri un nome specifico per l'operazione.
+
+
+### Operazioni Python
+
+Un' _Operazione Python_ è una funzione che, dati in input tensori e parametri, 
+crea una parte del grafo e restituisce tensori in output.
+
+*   I primi argomenti devono essere i tensori, seguiti dal paramentri base Python.
+    L'ultimo argomento è `name` con il valore di default `None`.
+*   Gli argomenti Tensori devono essere o un singolo tensore o un 'iterabile' di tensori. 
+    Perchè "Tensore o lista di Tensori" è troppo generico. Vedere `assert_proper_iterable`.
+*   Le operazioni che prendono tensori come argomenti, se stanno usando operazioni C++
+    devono chiamare `convert_to_tensor` per convertire input non-tensori in tensori.
+    Notare che nella documentazione gli argoemnti sono ancora descritti come un oggetto `Tensor` di uno
+    specifico dtype.
+*   Ogni operazione Python deve avere un `name_scope`. Come si può vedere sotto, passare il nome
+    dell'operazione come una stringa.
+*   Le operazioni devono contenere un commento Python estensibile con le dichiarazioni di 
+    Args and Returns, che spieghino il tipo e la semantica di ciascun valore. Eventuali
```

Comment:
Args and Returns?

Reply:
My approach to translation was to explain in Italian how to produce code fully understandable to every programmer in the world so, written in English.
Therefore, i managed Args and Returns as sections of the suggested coding standard, and I leaved them in English. Do you agree?

Reply:
I don't know.
Probably the untraslated Is Args e Returns without (and). 
But here I don't think it Is the reserverd syntax of the programming language but more the concepts of arguments and returns.
So strictly what Is the difference between talking about argomenti and ‘valori di ritorno’ and i.e. return as reserverd word of the programming lang?

Reply:
@mcaci What do you think?

Reply:
I agree with you, the sentence "...dichiarazioni di Args e Returns, che spieghino..." into explanation plain text sounds good. --> I'm going to update the document according to that...

To answer your question, in my opinion, we have three scenarios to deal with:
(a) reserved language tokens, i.e. 'return' in C/C++ --> no translation, no rewording, no capital initial
(b) strongly suggested tokens (part of the coding guidelines) that should be used in the code --> no translation, no rewording, capital initial i.e. 'Args' and 'Returns' 
(c) similar english words into plain text explanations, i.e. 'return values' meaning 'the values that are returned' -->  translated and maybe reworded

Reply:
Sorry for the delay, I have seen your comment now @bhack and @codeadmin-peritiae subsequent change and I have nothing to add, I agree with the fact that "Args e Returns" in the full context fits better than "Args and Returns"